modules/distributor/distributor.go

```diff
@@ -515,23 +567,53 @@ func requestsByTraceID(batches []*v1.ResourceSpans, userID string, spanCount int
 		traces = append(traces, r)
 	}
 
-	return keys, traces, nil
+	return keys, traces, tracesByID, nil
 }
 
-func recordDiscaredSpans(err error, userID string, spanCount int) {
-	s := status.Convert(err)
-	if s == nil {
-		return
-	}
-	desc := s.Message()
+func countDiscaredSpans(responses []*tempopb.IngesterPushResponse, tracesByID map[uint32]*rebatchedTrace, repFactor int) (int, int, int) {
+	maxLiveDiscardedCount := 0
+	traceTooLargeDiscardedCount := 0
+	internalErrorDiscardedCount := 0
+	discardedTraces := make(map[uint32]int)
+	numResponses := len(responses)
+	quorum := (repFactor / 2) + 1 // min success required
+
+	// max number of error allowed is {number of responses} - {minimum success required}
+	// + 1 for error count that should mark a trace as discarded
+	maxError := (numResponses - quorum) + 1
+
+	// if a trace triggers two different errors in two different responses, the error that triggers the maxError count will be used
+	for _, response := range responses {
+		if response == nil {
```

Comment:
i'm concerned about this error handling. if we get an error in DoBatch() above i'm fine with keeping it simple and just letting that error propagate all the way back just like it does now.